test/libgit2.jl

```diff
@@ -468,6 +468,65 @@ mktempdir() do dir
         #end
     #end
 
+    #@testset "Modify and reset repository" begin
+        repo = LibGit2.GitRepo(test_repo)
+        try
+            # check index for file
+            LibGit2.with(LibGit2.GitIndex(repo)) do idx
+                i = find(test_file, idx)
+                @test !isnull(i)
+                @test idx[get(i)] !== nothing
+            end
+
+            # check non-existent file status
+            st = LibGit2.status(repo, "XYZ")
+            @test isnull(st)
+
+            # check file status
+            st = LibGit2.status(repo, test_file)
+            @test !isnull(st)
+            @test LibGit2.isset(get(st), LibGit2.Consts.STATUS_CURRENT)
+
+            # modify file
+            open(joinpath(test_repo, test_file), "a") do io
+                write(io, 0x41)
+            end
+
+            # file modified but not staged
+            st_mod = LibGit2.status(repo, test_file)
+            @test !LibGit2.isset(get(st_mod), LibGit2.Consts.STATUS_INDEX_MODIFIED)
+            @test LibGit2.isset(get(st_mod), LibGit2.Consts.STATUS_WT_MODIFIED)
+
+            # stage file
+            LibGit2.add!(repo, test_file)
+
+            # modified file staged
+            st_stg = LibGit2.status(repo, test_file)
+            @test LibGit2.isset(get(st_stg), LibGit2.Consts.STATUS_INDEX_MODIFIED)
+            @test !LibGit2.isset(get(st_stg), LibGit2.Consts.STATUS_WT_MODIFIED)
+
+            # try to unstage to unknown commit
+            LibGit2.reset!(repo, "XYZ", test_file)
```

Comment:
I'm thinking of throwing exception when `revparse` (inside `reset!`) fails to find anything. At lease it would give some info about nature of failure.